src/components/Link/index.tsx

```diff
@@ -29,6 +31,71 @@ const Base = styled.a<InternalLinkProps>`
   }
 `;
 
+export class RouterLink extends React.Component {
+	constructor(props) {
+		super(props);
+
+		this.navigate = this.navigate.bind(this);
+	}
+
+	makeUrl() {
+		const { append, location, to } = this.props;
+
+		if (to) {
+			return toRelativeUrl(to);
+		}
+
+		if (append) {
+			return path.join(location.pathname, append);
+		}
+
+		return null;
+	}
+
+	navigate(event) {
+		const { blank, to } = this.props;
+
+		// If control or meta keys are pressed use default browser behaviour
+		if (event.ctrlKey || event.metaKey) {
+			return true;
+		}
+
+		// If the link is external use default browser behaviour
+		if (blank || isExternalUrl(to)) {
+			return true;
+		}
+
+		event.preventDefault();
+		const { history } = this.props;
+
+		const url = this.makeUrl();
+
+		history.push(url);
+		return false;
+	}
+
+	render() {
+		const props = omit(this.props, [
+			'match',
+			'location',
+			'history',
+			'to',
+			'append',
+		]);
+
+		const url = this.makeUrl();
+
+		return (
+			<Link
+				{...props}
+				href={url}
+				// We should only navigate when `url` is defined
+				onClick={url ? this.navigate : props.onClick}
+			/>
```

Comment:
I think we could just get rid of the `Link` component and implement it's (basic) functionality directly in `RouterLink`. That way we're keeping the component hierarchy a bit cleaner/simpler.